services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3339,6 +3338,30 @@ private OfflinePushStatusInfo getOffLineJobStatus(
               Optional.of(currentReturnStatusDetails.orElse("") + "Parent Kafka topic truncated");
         }
       }
+
+      // Update version status on parent
+      // COMPLETED -> ONLINE
+      // ERROR -> ERROR
+      // ERROR in some region but COMPLETED in others -> PARTIALLY_ONLINE
+      if (!incrementalPushVersion.isPresent()) {
```

Comment:
One more question is that Parent Controller may not be the SoT if someone starts a push job, but don't wait for it to be completed, and it means it doesn't poll the job status.
One use case is empty push via Admin Tool.

Maybe we could store a timestamp for version status update, and if someone is querying the version status of Parent Controller, and if the version status is not terminal and it hasn't been updated for some time, Parent Controller could collect the status from Child Colos on the fly and update the version status in Parent Controller.